pkg/provider/provider_test.go

```diff
@@ -466,6 +466,25 @@ func TestFormatKeyVaultObject(t *testing.T) {
 				ObjectAlias:    "alias",
 			},
 		},
+		{
+			desc: "no data loss for int properties",
+			keyVaultObject: KeyVaultObject{
+				ObjectName:           "secret1",
+				ObjectVersion:        "latest",
+				ObjectEncoding:       "base64",
+				ObjectType:           "secret",
+				ObjectAlias:          "alias",
+				ObjectVersionHistory: 12,
+			},
+			expectedKeyVaultObject: KeyVaultObject{
+				ObjectName:           "secret1",
+				ObjectVersion:        "latest",
+				ObjectEncoding:       "base64",
+				ObjectType:           "secret",
+				ObjectAlias:          "alias",
+				ObjectVersionHistory: 12,
+			},
+		},
```

Comment:
It'll be good to have more unit test coverage for:
1. the new function that returns the secret version list and 
2. the logic for filename when object version history is defined